File: C/lt_01_array/lt_01_array_04.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int main() {
  srand(time(NULL));
  int vetor[30], i, acima = 0, abaixo = 0, media = 0;

  for (i = 0; i < (sizeof(vetor) / sizeof(vetor[0])); i++) {
    vetor[i] = rand() % 100;
    printf("|%d", vetor[i]);
  }
  printf("|\n");

  for (i = 0; i < (sizeof(vetor) / sizeof(vetor[0])); i++) {
    media += vetor[i];
  }
  media /= (sizeof(vetor) / sizeof(vetor[0]));

  for (i = 0; i < (sizeof(vetor) / sizeof(vetor[0])); i++) {
    if (vetor[i] > media) {
      acima++;
    }
    if (vetor[i] < media) {
      abaixo++;
    }
  }

  printf("O valor da media e %d.\nA quantidade de maiores e %d.\nA "
         "quantidade de menores e %d.\n",
         media, acima, abaixo);
  return 0;
}

File: rust/lt01-array/lt01-array-04.rs
use std::time::{SystemTime, UNIX_EPOCH};

fn main() {
    let mut vetor = [0; 30];
    let media;
    let acima;
    let mut abaixo = vec![];
    //Calc media
    {
        let mut soma = 0;
        for i in 0..vetor.len() {
            vetor[i] = simple_random() % 101;
            soma += vetor[i];
        }
        media = soma as u32 / vetor.len() as u32;
        let mut aux = 0;
        for i in 0..vetor.len() {
            if vetor[i] > media {
                aux += 1;
            }
            if vetor[i] < media {
                abaixo.push(i as i32);
            }
        }
        acima = aux;
    }
    println!("O Vetor é {:?}", vetor);
    println!("A média dos grupos é {:.2}.", media);
    println!("A quantidade de valores acima da média é {}.", acima);
    println!("As posições de valores abaixo da média é {:?}.", abaixo);
}

fn simple_random() -> u32 {
    let seed = SystemTime::now()
        .duration_since(UNIX_EPOCH)
        .unwrap()
        .as_nanos() as u32; // Convert time to a u32 seed

    // A very basic pseudo-random generator (LCG)
    let mut rng = seed;
    rng = rng.wrapping_mul(1664525).wrapping_add(1013904223);
    rng
}
